vss_invoice_mgmt/src/pages/business/pages/masters/vendormaster/pages/VendorMasterList.jsx

```diff
@@ -0,0 +1,202 @@
+import React, { useEffect, useState } from "react";
+import AdminNavbar from "../../../../../../components/Navbars/AdminNavbar";
+import Sidebar from "../../../../../../components/Sidebar/Sidebar";
+import HeaderStats from "../../../../../../components/Headers/HeaderStats";
+import FooterAdmin from "../../../../../../components/Footers/FooterAdmin";
+import { TextField, Button } from "@mui/material";
+import { NavLink, useNavigate } from "react-router-dom";
+import {
+  GetVendorMasterList,
+  PutVendorMasterList,
+} from "../vendormasterconfig.js";
+import Stack from "@mui/material/Stack";
+import CircularProgres from "@/components/Circular-Progress/CircularProgress";
+import Pagination from "@mui/material/Pagination";
+import StatusSelect from "@/components/Status-Select/Status_Select";
+
+const VendorMasterList = () => {
+  const [searchName, setSearchName] = useState("");
+  const [searchContactNo, setSearchContactNo] = useState("");
+  const [searchStatus, setSearchStatus] = useState("");
+  const [vendormasterlist, setVendorMasterList] = useState([]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [loading, setLoading] = useState(false);
+  const [totalpages, settotalpages] = useState(1);
+  const itemsPerPage = 10;
+  const navigate = useNavigate();
+
+  const tableHead = {
+    col: [
+      "Sr No",
+      "Vendor Name",
+      "Vendor Contact No.",
+      "Status",
+      "Edit",
+      "Block",
+    ],
+  };
+
+  const getVendorList = async (page, name, contactNo, status) => {
+    setLoading(true);
+    try {
+      const users = await GetVendorMasterList(
+        1,
+        page,
+        itemsPerPage,
+        name,
+        contactNo,
+        status
+      );
+      setVendorMasterList(users.data);
+      settotalpages(Math.ceil(users.totalCount / itemsPerPage));
+    } catch (error) {
+      setVendorMasterList([]);
+      settotalpages(1);
+    }
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    if (searchName === "" && searchStatus === "") {
+      getVendorList(currentPage, "", "");
+    }
+  }, [searchName, searchStatus]);
```

Comment:
Why you use this?

Reply:
to show data back on screen after removing a data from search field